ambry-store/src/test/java/com.github.ambry.store/IndexSegmentTest.java

```diff
@@ -756,23 +757,56 @@ private void verifyReadFromFile(NavigableMap<MockId, NavigableSet<IndexValue>> r
     // read from file (unmapped) and verify that everything is ok
     Journal journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     IndexSegment fromDisk = createIndexSegmentFromFile(file, false, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, false, endOffset,
-        lastModifiedTimeInMs, resetKey);
-    verifyFind(referenceIndex, fromDisk);
-    verifyGetEntriesSince(referenceIndex, fromDisk);
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, false,
+        endOffset, lastModifiedTimeInMs, resetKey);
     // journal should contain all the entries
     verifyJournal(referenceIndex, journal);
     fromDisk.map(true);
 
     // read from file (mapped) and verify that everything is ok
     journal = new Journal(tempDir.getAbsolutePath(), Integer.MAX_VALUE, Integer.MAX_VALUE);
     fromDisk = createIndexSegmentFromFile(file, true, journal);
-    verifyIndexSegmentDetails(fromDisk, startOffset, numItems, expectedSizeWritten, true, endOffset,
+    verifyAllForIndexSegmentFromFile(referenceIndex, fromDisk, startOffset, numItems, expectedSizeWritten, true,
+        endOffset, lastModifiedTimeInMs, resetKey);
+    // journal should not contain any entries
+    assertNull("Journal should not have any entries", journal.getFirstOffset());
+
+    // delete the bloom file
+    File BloomFile = new File(file.getParent(),
+        IndexSegment.generateIndexSegmentFilenamePrefix(startOffset) + BLOOM_FILE_NAME_SUFFIX);
+    BloomFile.delete();
```

Comment:
either do `assertTrue("File could not be deleted", bloomFile.delete())` or use `FileUtils.delete()` (which throws an exception if delete fails)